Framework/LiveData/src/FileEventDataListener.cpp

```diff
@@ -148,6 +150,12 @@ std::shared_ptr<Workspace> FileEventDataListener::extractData() {
 
   m_runNumber = chunk->getRunNumber();
 
+  if (!m_loaderName.compare("LoadEventNexus")) {
+    // Scale the proton charge by the number of chunks
+    double charge = chunk->run().getProtonCharge();
```

Comment:
This should modify the individual values in the `proton_charge` log then call [`chunk->run(). integrateProtonCharge()`](https://github.com/mantidproject/mantid/blob/master/Framework/API/inc/MantidAPI/Run.h#L67) to get the correct integral value. That way the two logs (`proton_charge` and `gd_prtn_chrg`) will be consistent. It may have to be done by getting a copy of the log then setting it back on the run object.